### PR TITLE
Push hackathon

### PR DESCRIPTION
**Pull request by jpmedley**

### CODE REVIEW

File: src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md [hunk 1]
```diff
@@ -13,14 +13,22 @@ description: You've seen what makes a good notification. Now let's see how to im
   <img src="images/joe-asked.png" alt="The example notification.">
 </figure>
 
-Way back at the [beginning of this section](.), we
-showed a notification that looks like this and the code that goes with it.
+Way back at the [beginning of this article](.#anatomy), we
```
Comment (petele) [inline src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md]: change `(.#anatomy)` to `(#anatomy)`

File: src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md [hunk 2]
```diff
@@ -13,14 +13,22 @@ description: You've seen what makes a good notification. Now let's see how to im
...
 you enough information for it to be useful. That's what this section is about.
 
 <div style="clear:both;"></div>
 
+## Service workers, again
+
+Let's talk about service workers again. Handling message involves code that
+lives exclusively in a service worker. If you need a little background,
+[here's the introduction again](service-worker-primer). We've also got some
+handy instructions for [debugging service workers](../../../tools/chrome-devtools/debug/progressive-web-apps/#service-workers)
```
Comment (petele) [inline src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md]: change: `[debugging service workers](../../../tools/chrome-devtools/debug/progressive-web-apps/#service-workers)` to
`[debugging service workers](/web/tools/chrome-devtools/debug/progressive-web-apps/#service-workers)`

File: src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md (hunk 1 above)
Comment (jpmedley) [inline src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md]: Fixed.

File: src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md (hunk 2 above)
Comment (jpmedley) [inline src/content/en/fundamentals/engage-and-retain/push-notifications/handling-messages.md]: Fixed.

**Comment by gauntface** [comment]
I think we are good to merge.

**Comment by PaulKinlan** [comment]
@petele went through this, it looked great.  VAPID bit and application key is great!

**Pull request merged**